### PR TITLE
[docs] Optimizer and integrations

### PR DESCRIPTION
**Pull request by stevhliu**
This PR cleans up the optimizer section by separating the actionable and practical code examples from the more conceptual explanations of how 8-bit optimizers work. I also added very brief code examples in the Integration section to better show how they work for each HF library. Other more minor things include:

➖ remove duplicate Quantization doc from the API section
➕ new explanation doc for how 8-bit optimizers work

cc @Titus-von-Koeller

### CODE REVIEW

File: docs/source/index.mdx [hunk 1]
```diff
@@ -1,19 +1,13 @@
...
-- [Integrations](integrations)
-- [Optimizers](optimizers)
+* 8-bit optimizers uses block-wise quantization to maintain 32-bit performance at a small fraction of the memory cost.
+* LLM.Int() or 8-bit quantization enables large language model inference with only half the required memory and without any performance degradation. This method is based on vector-wise quantization to quantize most features to 8-bits and separately treating outliers with 16-bit matrix multiplication.
+* QLoRA or 4-bit quantization enables large language model training with several memory-saving techniques that don't compromise performance. This method quantizes a model to 4-bits and inserts a small set of trainable low-rank adaptation (LoRA) weights to allow training.
 
 # License
 
-The majority of bitsandbytes is licensed under MIT, however portions of the project are available under separate license terms, as the parts adapted from Pytorch are licensed under the BSD license.
+The majority of bitsandbytes is MIT licensed. However, portions of the project are available under separate license terms such as the parts adapted from PyTorch which are licensed under the BSD license.
```
Comment (matthewdouglas) [inline docs/source/index.mdx:11]: I'm a little curious - do we know specifically which parts are adapted from PyTorch?

Comment (stevhliu) [reply docs/source/index.mdx:11]: Sorry, I'm not too sure which parts are specifically adapted from PyTorch. Still finding my way around the library 🙂 

I think @Titus-von-Koeller or @younesbelkada can help answer?

Comment (Titus-von-Koeller) [reply docs/source/index.mdx:11]: Hey! Yeah, I already pinged Tim about ti. Also about the licensing stuff in general. From what I understand it's only 8 lines of code from PyTorch but I don't know which, yet. I'll get back to you on this.

File: docs/source/integrations.mdx
Comment (Titus-von-Koeller) [inline docs/source/integrations.mdx:1]: I really like the refactor you did in this section. Everything flows much better, is coherent and I like the new additions, cleaned up code examples and clarifications a lot! Great work, thanks :D I wish I had these docs when I got started with BNB at the time 😅

File: docs/source/optimizers.mdx [hunk 2]
```diff
@@ -178,13 +89,6 @@ class MyModule(torch.nn.Module):
...
-... under construction ...
-
-Here we'll provide further auto-generated API docs soon. Please feel free to contribute doc-strings for the respective optimizers, as `bitsandbytes` is a community effort.
-
-### StableEmbedding[[stable-emb-api]]
+## Next steps
 
-[[autodoc]] bitsandbytes.nn.StableEmbedding
-    - __init__
+For more conceptual details and explanation about 8-bit optimizers, take a look at the [8-bit optimizers](./explanations/optimizers) guide.
```
Comment (Titus-von-Koeller) [inline docs/source/optimizers.mdx:94]: Really like the new structure, much more graspable and gradual disclosure of details is dosed just right. Excellent!

File: docs/source/optimizers.mdx
Comment (Titus-von-Koeller) [inline docs/source/optimizers.mdx:1]: Nice rework, 0 complaints!

**Review by Titus-von-Koeller — APPROVED** [review]

**Pull request merged**